diff -r --unified SDL2-2.30.3-orig/configure.ac SDL2-2.30.3/configure.ac
--- SDL2-2.30.3-orig/configure.ac	2024-05-02 02:16:27.000000000 +1000
+++ SDL2-2.30.3/configure.ac	2024-05-12 00:46:48.067818578 +1000
@@ -1922,6 +1922,7 @@
                 fi
                 ;;
         esac
+        LIBS="$LIBS -lxcb -lXau"
         AC_PATH_X
         AC_PATH_XTRA
         if test x$have_x = xyes; then
diff -r --unified SDL2-2.30.3-orig/include/SDL_config.h.in SDL2-2.30.3/include/SDL_config.h.in
--- SDL2-2.30.3-orig/include/SDL_config.h.in	2024-05-02 02:16:27.000000000 +1000
+++ SDL2-2.30.3/include/SDL_config.h.in	2024-05-12 00:39:40.427756760 +1000
@@ -199,7 +199,7 @@
 #undef HAVE_SETJMP
 #undef HAVE_NANOSLEEP
 #undef HAVE_SYSCONF
-#undef HAVE_SYSCTLBYNAME
+#undef HAVE_SYSCTLBYNAME_BLOCK
 #undef HAVE_CLOCK_GETTIME
 #undef HAVE_GETPAGESIZE
 #undef HAVE_MPROTECT
diff -r --unified SDL2-2.30.3-orig/sdl2.pc.in SDL2-2.30.3/sdl2.pc.in
--- SDL2-2.30.3-orig/sdl2.pc.in	2024-05-02 02:16:27.000000000 +1000
+++ SDL2-2.30.3/sdl2.pc.in	2024-05-12 00:39:40.427756760 +1000
@@ -10,5 +10,5 @@
 Version: @SDL_VERSION@
 Requires.private: @PKGCONFIG_DEPENDS@
 Conflicts:
-Libs: -L${libdir} @SDL_RLD_FLAGS@ @SDL_LIBS@ @PKGCONFIG_LIBS_PRIV@ @SDL_STATIC_LIBS@
+Libs: -L${libdir}  -lSDL2   -lm  -lX11 -lXext -lxcb -lXau -lpthread
 Cflags: -I${includedir} -I${includedir}/SDL2 @SDL_CFLAGS@
diff -r --unified SDL2-2.30.3-orig/src/thread/pthread/SDL_systhread.c SDL2-2.30.3/src/thread/pthread/SDL_systhread.c
--- SDL2-2.30.3-orig/src/thread/pthread/SDL_systhread.c	2024-05-02 02:16:27.000000000 +1000
+++ SDL2-2.30.3/src/thread/pthread/SDL_systhread.c	2024-05-12 00:39:40.427756760 +1000
@@ -29,6 +29,7 @@
 #include <pthread_np.h>
 #endif
 
+#include <sched.h>
 #include <signal.h>
 #include <errno.h>
 
