diff -r --unified libxcb-1.17.0-orig/src/config.h.in libxcb-1.17.0/src/config.h.in
--- libxcb-1.17.0-orig/src/config.h.in	2024-04-16 01:09:34.000000000 +1000
+++ libxcb-1.17.0/src/config.h.in	2024-05-11 22:31:57.196663200 +1000
@@ -28,7 +28,7 @@
 #undef HAVE_MINIX_CONFIG_H
 
 /* Define if your platform supports sendmsg */
-#undef HAVE_SENDMSG
+#undef HAVE_SENDMSG_BLOCK
 
 /* Have the sockaddr_un.sun_len member. */
 #undef HAVE_SOCKADDR_SUN_LEN
